src/client/Microsoft.Identity.Client/Internal/ClientCredential/IClientCredential.cs

```diff
@@ -11,12 +11,15 @@
 using Microsoft.Identity.Client.Core;
 using Microsoft.Identity.Client.OAuth2;
 using Microsoft.Identity.Client.PlatformsCommon.Interfaces;
+using Microsoft.Identity.Client.TelemetryCore;
 using Microsoft.Identity.Client.Utils;
 
 namespace Microsoft.Identity.Client.Internal.ClientCredential
 {
     internal interface IClientCredential
     {
+        AssertionType AssertionType { get; }
```

Comment:
This is a code smell. You already know the type by looking at the `obj.Type` or `typeof(obj)` where obj implements `IClientCredential`. We should as much as possible not complicate the object model for telemetry.